Plugin~/WebRTCPlugin/Context.cpp

```diff
@@ -174,80 +164,96 @@ namespace WebRTC
         clients.clear();
         peerConnectionFactory = nullptr;
         audioTrack = nullptr;
-        videoTracks.clear();
-        audioStream = nullptr;
-        videoStreams.clear();
+        mediaSteamTrackList.clear();
+        mediaStreamMap.clear();
+        m_mapSetSessionDescriptionObserver.clear();
+        videoCapturerList.clear();
 
         workerThread->Quit();
         workerThread.reset();
         signalingThread->Quit();
         signalingThread.reset();
     }
 
-    bool Context::InitializeEncoder(IGraphicsDevice* device)
+    bool Context::InitializeEncoder(IGraphicsDevice* device, webrtc::MediaStreamTrackInterface* track)
     {
-        if(!nvVideoCapturer->InitializeEncoder(device, m_encoderType))
+        if(videoCapturerList[track]->InitializeEncoder(device, m_encoderType))
         {
-            return false;
+            videoCapturerList[track]->StartEncoder();
+            return true;
         }
-        nvVideoCapturer->StartEncoder();
-        return true;
+        return false;
     }
-    void Context::EncodeFrame()
+    void Context::EncodeFrame(webrtc::MediaStreamTrackInterface* track)
     {
-        nvVideoCapturer->EncodeVideoData();
+        if (!videoCapturerList[track]->EncodeVideoData())
+        {
+        }
     }
-    void Context::FinalizeEncoder()
+    void Context::FinalizeEncoder(webrtc::MediaStreamTrackInterface* track)
     {
-        nvVideoCapturer->FinalizeEncoder();
+        videoCapturerList[track]->FinalizeEncoder();
     }
 
     UnityEncoderType Context::GetEncoderType() const
     {
         return m_encoderType;
     }
 
-    webrtc::MediaStreamInterface* Context::CreateVideoStream(void* frameBuffer, int width, int height)
+    webrtc::MediaStreamInterface* Context::CreateMediaStream(const std::string& streamId)
     {
-        rtc::scoped_refptr<webrtc::VideoTrackSourceInterface> source(WebRTC::VideoCapturerTrackSource::Create(workerThread.get(), std::move(nvVideoCapturerUnique), false));
+        auto stream = peerConnectionFactory->CreateLocalMediaStream(streamId);
+        const auto observer = new MediaStreamObserver(stream);
+        stream->RegisterObserver(observer);
+        m_mapMediaStreamObserver[stream] = observer;
+        return stream.release();
```

Comment:
Yes, we need to release the instance manually because the method returns the pointer without destruction instance.

I fixed the issue that remaining unreleased objects.
https://github.com/Unity-Technologies/com.unity.webrtc/pull/81/files#diff-3aa0908b7261cc012e588c33f38e09c8R222